core/new-gui/src/app/workspace/types/operator-schema.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * This file contains multiple type declarations related to operator schema.
+ * These type declarations should be the same with the backend API.
+ */
+
+export interface OperatorAdditionalMetadata {
+    userFriendlyName: string;
+    numInputPorts: number;
+    numOutputPorts: number;
+    advancedOptions?: string[];
+    operatorGroupName?: string;
+    operatorDescription?: string;
+}
+
+export interface OperatorSchema {
+    operatorType: string;
+    jsonSchema: Object;
+    additionalMetadata: OperatorAdditionalMetadata;
+}
+
+export interface GroupInfo {
+    groupName: string;
```

Comment:
In case of such definitions, which serve as a protocol with server, please give an example of what is expected from server and explain.

Reply:
it's in another mock data file. Comments are added to point to that file.